File: testetest/employee-management-system/src/main/java/com/dailycodebuffer/employee/services/RentalService.java
package com.dailycodebuffer.employee.services;

import com.dailycodebuffer.employee.model.Rental;
import com.dailycodebuffer.employee.model.Item;
import com.dailycodebuffer.employee.model.RentalHistory;

import java.util.List;

public interface RentalService {

    Rental createRental(Rental employee);

    RentalHistory createRentalHistory(RentalHistory rentalHistory);

    List<Rental> getAllRentals();

    boolean deleteRental(Long id);

    Rental getRentalById(Long id);

    Item getItemById(String id);

    Rental updateRental(Long id, Rental employee);

    Rental getRentalByItemCode(String code);
}

File: testetest/src/services/RentalService.js
import axios from "axios";

const RENTAL_API_BASE_URL = "http://localhost:8080/api/v1/rentals";
const HISTORY_API_BASE_URL = "http://localhost:8080/api/v1/history";
const ITEMS_API_BASE_URL = "http://localhost:8080/api/v1/items";

class RentalService {
  getRentalByItemCode(code) {
    return axios.get(RENTAL_API_BASE_URL + "/code/" + code);
  }

  getItemById(id) {
    return axios.get(ITEMS_API_BASE_URL + "/" + id);
  }

  saveRental(rental) {
    return axios.post(RENTAL_API_BASE_URL, rental);
  }

  historyRental(rental){
    return axios.post(HISTORY_API_BASE_URL, rental);
  }

  getRentals() {
    return axios.get(RENTAL_API_BASE_URL);
  }

  deleteRental(id) {
    return axios.delete(RENTAL_API_BASE_URL + "/" + id);
  }

  getRentalById(id) {
    return axios.get(RENTAL_API_BASE_URL + "/" + id);
  }

  updateRental(rental, id) {
    return axios.put(RENTAL_API_BASE_URL + "/" + id, rental);
  }
}

export default new RentalService();
